### PR TITLE
add theme nightfox

### PR DESCRIPTION
**Pull request by jhscheer**
This theme is an adaptation of
[github.com/EdenEast/nightfox.nvim](https://github.com/EdenEast/nightfox.nvim)

helix_theme_nightfox

### CODE REVIEW

File: runtime/themes/nightfox.toml [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+"punctuation" = { fg = "fg2" } # Any punctuation symbol.
+"punctuation.delimiter" = { fg = "fg2" } # Commas, colons or other delimiter depending on the language.
+"punctuation.bracket" = { fg = "fg2" } # Parentheses, angle brackets, etc.
+"punctuation.special" = { fg = "fg2" } # String interpolation brackets
+
+"keyword" = { fg = "magenta" } # Language reserved keywords.
+"keyword.control" = { fg = "magenta-bright" } # Control keywords.
+"keyword.control.conditional" = { fg = "magenta-bright" } # `if`, `else`, `elif`.
+"keyword.control.repeat" = { fg = "magenta-bright" } # `for`, `while`, `loop`.
+"keyword.control.import" = { fg = "magenta-bright" } # `import`, `export` `use`.
```
Comment (archseer) [inline runtime/themes/nightfox.toml:130]: You don't need all these additional definitions, `keyword.control.conditional` will fallback onto `keyword.control` and so on

File: runtime/themes/nightfox.toml [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+"constructor" = { fg = "magenta"} # Constructor method for a class or struct.
+
+"constant" = { fg = "yellow" } # Constant value
+"constant.builtin" = { fg = "orange-bright" } # Special constants like `true`, `false`, `none`, etc.
+"constant.builtin.boolean" = { fg = "orange" } # True or False.
+"constant.character" = { fg = "green" } # Constant of character type.
+"constant.character.escape" = { bold = true, fg = "yellow-bright" } # escape codes like \n.
+"constant.numeric"  = { fg = "orange" } # constant integer or float value.
+"constant.numeric.integer" = { fg = "orange" } # constant integer value.
+"constant.numeric.float" = { fg = "orange" } # constant float value.
```
Comment (archseer) [inline runtime/themes/nightfox.toml:99]: Same here

File: runtime/themes/nightfox.toml [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+"variable.parameter" = { fg = "cyan-bright" } # Function parameters.
+# "variable.other" = { }
+"variable.other.member" = { fg = "fg2" } # Fields of composite data types (e.g. structs, unions).
+
+"label" = { fg = "magenta-bright" } # Loop labels, among other things.
+
+"punctuation" = { fg = "fg2" } # Any punctuation symbol.
+"punctuation.delimiter" = { fg = "fg2" } # Commas, colons or other delimiter depending on the language.
+"punctuation.bracket" = { fg = "fg2" } # Parentheses, angle brackets, etc.
+"punctuation.special" = { fg = "fg2" } # String interpolation brackets
```
Comment (archseer) [inline runtime/themes/nightfox.toml:124]: And here

**Comment by workingj** [comment]
> This theme is an adaptation of [github.com/EdenEast/nightfox.nvim](https://github.com/EdenEast/nightfox.nvim)
> 
> helix_theme_nightfox

What font is this ?

thx

**Comment by jhscheer** [comment]
@workingj
`alacritty.yml`:
```yml
family: DroidSansMono Nerd Font
```

File: runtime/themes/nightfox.toml [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+# "ui.virtual" = { } # Namespace for additions to the editing area.
+"ui.virtual.ruler" = { bg = "bg3" } # Vertical rulers (colored columns in editing area).
+"ui.virtual.whitespace" = { fg = "bg3" } # Whitespace markers in editing area.
+"ui.virtual.indent-guide" = { fg = "black" } # Vertical indent width guides
+
+"ui.statusline" = { fg = "fg2", bg = "bg0" } # Status line.
+"ui.statusline.inactive" = { fg = "fg3", bg = "bg0" } # Status line in unfocused windows.
+"ui.statusline.normal" = { bg = "cyan", bold = true, fg = "bg0" } # Statusline mode during normal mode (only if editor.color-modes is enabled)
+"ui.statusline.insert" = { bg = "green", bold = true, fg = "bg0" } # Statusline mode during insert mode (only if editor.color-modes is enabled)
+"ui.statusline.select" = { bg = "magenta", bold = true, fg = "bg0" } # Statusline mode during select mode (only if editor.color-modes is enabled)
```
Comment (the-mikedavis) [inline runtime/themes/nightfox.toml:43]: these should be `modifiers = ["bold"]` instead

File: runtime/themes/nightfox.toml [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+# "markup.normal.completion" = { } # For completion doc popup ui.
+# "markup.normal.hover" = { } # For hover popup ui.
+
+# "markup.heading.completion" = { } # Headings for completion doc popup ui.
+# "markup.heading.hover" = { } # Headings for hover popup ui.
+
+# "markup.raw" = { } # Code block in Markdown.
+# "markup.raw.inline" = { } # `Inline code block` in Markdown.
+# "markup.raw.inline.completion" = { } # For completion doc popup ui
+# "markup.raw.inline.hover" = { } # For hover popup ui
```
Comment (the-mikedavis) [inline runtime/themes/nightfox.toml:64]: You may want to implement the first or the first two - these are used in Markdown documents in code fences and inline-code. The other two scopes don't exist

Comment (jhscheer) [reply runtime/themes/nightfox.toml]: Done.
Thanks for your review.

File: runtime/themes/nightfox.toml [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+"type.builtin" = { fg = "cyan-bright" } # Primitive types of the language (string, int, float).
+"type.enum.variant" = { fg = "orange" }
+
+"constructor" = { fg = "magenta" } # Constructor method for a class or struct.
+
+"constant" = { fg = "yellow" } # Constant value
+"constant.builtin" = { fg = "orange-bright" } # Special constants like `true`, `false`, `none`, etc.
+"constant.builtin.boolean" = { fg = "orange" } # True or False.
+"constant.character" = { fg = "green" } # Constant of character type.
+"constant.character.escape" = { bold = true, fg = "yellow-bright" } # escape codes like \n.
```
Comment (the-mikedavis) [inline runtime/themes/nightfox.toml:87]: This one should be `modifiers = ["bold"]` as well

File: runtime/themes/nightfox.toml [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+# "diagnostic.hint" = { } #	Diagnostics hint (editing area)
+# "diagnostic.info" = { } #	Diagnostics info (editing area)
+# "diagnostic.warning" = { } #	Diagnostics warning (editing area)
+# "diagnostic.error" = { } #	Diagnostics error (editing area)
+
+
+# SYNTAX HIGHLIGHTING
+# These keys match tree-sitter scopes.
+
+"property" = { fg = "blue" } # Regex group names.
```
Comment (the-mikedavis) [inline runtime/themes/nightfox.toml:73]: IIRC we removed this scope, it doesn't have an effect anymore

File: runtime/themes/nightfox.toml [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+
+"ui.menu" = { bg = "sel0", fg = "fg1" } # Code and command completion menus.
+"ui.menu.selected" = { bg = "fg3" } # Selected autocomplete item.
+"ui.menu.scroll"	= { fg = "fg3" } # fg sets thumb color, bg sets track color of scrollbar.
+
+"ui.popup" = { bg = "bg0", fg = "fg1" } # Documentation popups (space-k).
+"ui.popup.info" = { bg = "sel0", fg = "fg1" } # Info popups box (space mode menu).
+
+"markup.raw" = { fg = "magenta" } # Code block in Markdown.
+"markup.raw.inline" = { fg = "orange", modifiers = ["bold"] } # `Inline code block` in Markdown.
```
Comment (the-mikedavis) [inline runtime/themes/nightfox.toml:55]: There are a few other markup scopes that are very common to add. The full list is here https://github.com/helix-editor/helix/blob/0e23e4f882ec840bed0e2c1ef3723deb7a258801/book/src/themes.md?plain=1#L205

at least `markup.heading`, `markup.bold`, `markup.italic`, and `markup.link.url` and `markup.link.text` should be added or else markdown documents will not be highlighted very much

**Comment by jhscheer** [comment]
I noticed, that some keys listed in `book/src/themes.md` don't work like I expected,
e.g.
```toml
"keyword.control.repeat" = { fg = "pink" } # `for`, `while`, `loop`.
"comment.block.documentation" = { fg = "comment", modifiers = ["bold"] } # Doc comments, e.g '///' in rust.
```
I want to change the color of `for` in `*.rs`, but it's always the color of of `keyword` and `keyword.control.repeat` is ignored.
Same with `comment.block.documentation`, I would like it to be bold, but it is ignored.

Is this a (known) bug, or is my understanding flawed?

**Comment by the-mikedavis** [comment]
Not all languages use those captures consistently. For example the Rust highlights don't capture doc comments as `comment.block.documentation` (despite the docs using it as an example) but some other languages do.

`for` is captured as `keyword.control.repeat` https://github.com/helix-editor/helix/blob/46eb185d45c22f95de36b8467f323247cdb76fda/runtime/queries/rust/highlights.scm#L113-L114

This was changed somewhat recently though (https://github.com/helix-editor/helix/pull/4510) so you will need to have built from source rather than use the 22.08.1 release to see that capture work.

File: runtime/themes/nightfox.toml [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+
+# DIAGNOSTICS
+"warning" = { fg ="yellow", bg = "bg1" } # Diagnostics warning (gutter)
+"error" = { fg = "red", bg = "bg1" } # Diagnostics error (gutter)
+"info" = { fg = "blue", bg = "bg1" } # Diagnostics info (gutter)
+"hint" = { fg = "green", bg = "bg1" } # Diagnostics hint (gutter)
+"diagnostic" = { modifiers = ["underlined"] } # Diagnostics fallback style (editing area)
+"diagnostic.error" = { fg = "red" } #	Diagnostics error (editing area)
+"diagnostic.warning" = { fg = "yellow" } #	Diagnostics warning (editing area)
+"diagnostic.info" = { fg = "info" } #	Diagnostics info (editing area)
```
Comment (the-mikedavis) [inline runtime/themes/nightfox.toml:70]: it looks like this is not defined in the palette

Comment (jhscheer) [reply runtime/themes/nightfox.toml]: good eye. fixed it. sorry for that.

**Review by the-mikedavis — APPROVED** [review]

**Comment by the-mikedavis** [comment]
Could you add a screenshot of this to the themes wiki? https://github.com/helix-editor/helix/wiki/Themes

**Comment by jhscheer** [comment]
> Could you add a screenshot of this to the themes wiki? https://github.com/helix-editor/helix/wiki/Themes

done

**Pull request merged**